distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Match_2.enso

```diff
@@ -53,15 +62,176 @@ type Match_2
+    grapheme_span self id=0 default=Nothing =
+        result = self.span id default
+        case result == default of
+            True -> default
+            False -> case result of
+                Nothing -> Nothing
+                utf_16_span : Utf_16_Span -> utf_16_span.to_grapheme_span
+
+    ## Gets the Text matched by the group with the provided identifier, or
+       a default value if the group did not participate in the match. If no such group
+       exists for the provided identifier, a `No_Such_Group` is returned.
+
+       Arguments:
+       - id: The integer index or name of that group.
+
+       ? The Full Match
+         The group with index 0 is always the full match of the pattern.
+
+       ? Named Groups by Index
+         If the regex contained named groups, these may also be accessed by
+         index based on their position in the pattern.
+
+       Note that it is possible for a group to "not participate in the match",
+       for example with a disjunction. In the example below, the "(d)" group
+       does not participate -- it neither matches nor fails.
+
+            "ab((c)|(d))".find "abc"
+
+       In this case, the group id for "(d)", which is 3, is a valid group id and
+       (Pattern_2.lookup_group 3) will return 3. If the caller tries to get group 3,
+       Match_2.group will return the default value.
+    group : Integer | Text -> Any | Nothing -> Text ! No_Such_Group
+    group self id default=Nothing =
+        result = self.grapheme_span id default
+        case result == default of
+            True -> default
+            False -> case result of
+                Nothing -> Nothing
+                span -> span.text
+
+    ## Gets the Text matched by the entire pattern.
+    text self = self.group 0
+
+    ## Gets a vector containing the Text of _all_ of the capturing groups in
+       the pattern, replacing the value of groups that did not participate in
+       the match with `default`. This vector includes group 0, which contains
+       the entire match.
+
+       Arguments:
+       - default: The value to return for a given index when the group at that
+         index did not participate in the match.
+
+       ? The Full Match
+         The group with index 0 is always the full match of the pattern.
+
+       ? Named Groups by Index
+         If the regex contained named groups, these may also be accessed by
+         index based on their position in the pattern.
+
+       > Example
+         Get a vector of the text matched by all of the groups in this match,
+         replacing the value for groups that didn't match with "UNMATCHED".
+
+             import Standard.Examples
+
+             example_groups =
+                 match = Examples.match
+                 match.groups default="UNMATCHED"
+    groups : Any -> Vector (Text | Any)
+    groups self default=Nothing =
+        group_numbers = 0.up_to self.pattern.group_count
```

Comment:
this will drop the last group unless I am mistaken - needs a `+1`

Reply:
group_count is 5 in this case, it includes group 0.
This is verified at https://github.com/enso-org/enso/blob/wip/gmt/5122-replace/test/Tests/src/Data/Text/Regex_2_Spec.enso#L421
(which will be reactivated in the next PR)